messages/src/main/java/no/unit/nva/publication/messages/delete/DeleteMessageHandler.java

```diff
@@ -22,35 +29,73 @@ public class DeleteMessageHandler extends ApiGatewayHandler<Void, Void> {
     private final MessageService messageService;
+    private final TicketService ticketService;
+    private final ResourceService resourceService;
 
     @JacocoGenerated
     public DeleteMessageHandler() {
-        this(MessageService.defaultService());
+        this(MessageService.defaultService(), TicketService.defaultService(), ResourceService.defaultService());
     }
 
-    public DeleteMessageHandler(MessageService messageService) {
+    public DeleteMessageHandler(MessageService messageService, TicketService ticketService,
+                                ResourceService resourceService) {
         super(Void.class);
         this.messageService = messageService;
+        this.ticketService = ticketService;
+        this.resourceService = resourceService;
     }
 
     @Override
-    protected void validateRequest(Void unused, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+    protected void validateRequest(Void unused, RequestInfo requestInfo, Context context) {
         //Do nothing
     }
 
     @Override
     protected Void processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
 
         var userInstance = UserInstance.fromRequestInfo(requestInfo);
-
-        attempt(() -> extractMessageIdentifier(requestInfo))
-            .map(this::fetchMessage)
+        attempt(() -> extractMessageIdentifier(requestInfo)).map(this::fetchMessage)
             .forEach(message -> messageService.deleteMessage(userInstance, message))
             .orElseThrow(this::mapException);
 
+        var ticketIdentifier = getTicketIdentifier(requestInfo);
+        completeTicketWhenGeneralSupportWithNoActiveMessages(ticketIdentifier, userInstance);
+
         return null;
     }
 
+    private static SortableIdentifier getTicketIdentifier(RequestInfo requestInfo) {
+        return new SortableIdentifier(requestInfo.getPathParameter(TICKET_IDENTIFIER_PATH_PARAMETER));
+    }
+
+    @Override
+    protected Integer getSuccessStatusCode(Void input, Void output) {
+        return HTTP_OK;
+    }
+
+    private static boolean allMessagesAreDeleted(List<Message> messages) {
+        return messages.stream().allMatch(message -> DELETED.equals(message.getStatus()));
+    }
+
+    private void completeTicketWhenGeneralSupportWithNoActiveMessages(SortableIdentifier ticketIdentifier,
+                                                                      UserInstance userInstance)
+        throws NotFoundException {
+        var ticket = ticketService.fetchTicketByIdentifier(ticketIdentifier);
+        if (ticket instanceof GeneralSupportRequest) {
+            var messages = ticket.fetchMessages(ticketService);
+            if (allMessagesAreDeleted(messages)) {
+                var resource = getResource(ticket.getResourceIdentifier());
+                ticket.complete(resource.toPublication(), userInstance).persistUpdate(ticketService);
+            }
+        }
+    }
```

Comment:
It would be nice if we could fetch both tickets and messages by ticketIdentifier by single database transaction, but it is not possible.